sdk/iOS/src/MSSyncTable.h

```diff
@@ -98,17 +98,17 @@
 /// MSQeury object.
 /// Before a pull is allowed to run, all pending requests on the specified table will be sent to
 /// the server. If a pending request for this table fails, the pull will be cancelled
--(void)pullWithQuery:(MSQuery *)query queryId:(NSString *)queryId completion:(MSSyncBlock)completion;
+-(void)pullWithQuery:(nonnull MSQuery *)query queryId:(nullable NSString *)queryId completion:(nullable MSSyncBlock)completion;
```

Comment:
For the queryId when passed to `-[MSSyncContext pullWithQuery:queryId:completion:]` you do a heap of checks for `if (queryId)` hence I assumed it could be nullable.